src/services/validation.service.js

```diff
@@ -1,93 +1,112 @@
+    },
+    min: function (value, $scope, attr) {
+        return parseFloat(value) >= parseFloat(attr.min);
+    },
+    max: function (value, $scope, attr) {
+        return parseFloat(value) <= parseFloat(attr.max);
+    },
+    length: function (value, $scope, attr) {
+        return value.length === parseInt(attr.length);
+    },
+    strictemail: function(value, $scope, attr) {
+      return  new RegExp(/^[_a-z0-9]+(\.[_a-z0-9]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,4})$/).test(value);
+    }
+  };
 
-    var selectors = [];
-    var elements = ["input", "select", "textarea"];
+  var selectors = [];
+  var elements = ['input', 'select', 'textarea'];
 
-    angular.forEach(elements, function (element) {
-        selectors.push(element + "[ng-model]");
-        selectors.push(element + "[data-ng-model]");
-    });
+  angular.forEach(elements, function (element) {
+    selectors.push(element + '[ng-model]');
+    selectors.push(element + '[data-ng-model]');
+  });
 
-    var selector = selectors.join(",");
+  var selector = selectors.join(',');
 
-    return {
-        /**
-         * Search all the input element inside the given DOM element and apply the "bs-validation" directive so we
-         * need not a add it for every form element.
-         * @param $element
-         */
-        addDirective: function ($element) {
-            var validateableElements = $element.findAll(selector);
-            validateableElements.attr("bs-validation", "");
-            return validateableElements;
-        },
-        addToNgIncludedURLs: function (url) {
-            if (ngIncludedURLs.indexOf(url) === -1) {
-                ngIncludedURLs.push(url);
-            }
-        },
-        addValidator: function ($scope, $attr, ngModelController, validatorKey) {
-            ngModelController.$validators[validatorKey] = function (modelValue, viewValue) {
-                var value = modelValue || viewValue;
-                return ngModelController.$isEmpty(value) || genericValidators[validatorKey](value, $scope, $attr);
-            };
-        },
-        checkNgIncludedURL: function (url) {
-            var index = ngIncludedURLs.indexOf(url);
-            if (index > -1) {
-                ngIncludedURLs.splice(index, 1);
-                return true;
-            }
+  return {
+    /**
+     * Search all the input element inside the given DOM element and apply the 'bs-validation' directive so we
+     * need not a add it for every form element.
+     * @param $element
+     */
+    getValidators: function (){
+      var builtIn = ['equalto', 'min', 'max', 'number', 'digits', 'length']
+      var additional = Object.keys(genericValidators);
+      return builtIn.concat(additional);
+    },
 
-            return false;
-        },
-        getDefaultMessage: function (key) {
-            return messages[key];
-        }
-    };
-});
\ No newline at end of file
+    getMeta: function (){
+      return ['matchName'];
```

Comment:
If you have a password matching like this:

```html
<form name="form" ng-submit="su.signUp()" novalidate>

          <div class="form-group">
            <input type="email"
              class="form-control"
              name="email"
              placeholder="Email"
              ng-model="su.user.email"

              bs-validation
              strictemail="true"
              required="true"
              response-notification="{{su.error.email}}">

          </div>

          <div class="form-group">
            <input type="password"
              class="form-control"
              name="password"
              placeholder="Password"
              ng-model="su.user.password"

              bs-validation
              ng-minlength="3"
              required="true"
              response-notification="{{su.error.password}}">

          </div>

          <div class="form-group">
            <input type="password"
              class="form-control"
              name="password_confirmation"
              placeholder="Password Confirmation"
              ng-model="su.user.password_confirmation"

              bs-validation
              equalTo="{{su.user.password}}"
              matchName="password"
              required="true"
              response-notification="{{su.error.password_confirmation}}">
          </div>

          <p>
            already an admin? <a ui-sref="main.auth.signin">sign in!</a>
          </p>
          <div class="control-group">
            <div class="controls">
              <button ng-disabled="form.$invalid" type="submit" class="btn btn-primary pull-right">sign up</button>
            </div>
          </div>
        </form>
```

you don´t want to have `must match {{whats in the password input}}` but rather `must match password`

so you need something which is the meta information for 'password' in this case.